Source/Dafny/Compilers/Compiler.cs

```diff
@@ -1238,8 +1242,10 @@ public enum ResolvedUnaryOp { BoolNot, BitwiseNot, Cardinality }
             }
           } else if (d is ValuetypeDecl) {
             // nop
+            wr.DeleteLast();
```

Comment:
That's a bit scary: are we sure that we know which character this will be deleting?

Reply:
Reasonably so, the test go trough after all. At least in this case it should be.

Reply:
I agree this is very risky and makes the code an order of magnitude harder to reason about confidently. I'd define whatever additional `Compiler` attributes you need to specialize the behavior without doing this.

I hesitate to even mention it, but IF this was absolutely necessary, I would at least provide an argument to `DeleteLast` to specify WHAT you expect to be deleted, so at least you have a chance of catching bugs if the state is ever not what you thought it would be.

Reply:
I think it's reasonably nice now.